src/components/__tests__/PropsTable.test.tsx

```diff
@@ -0,0 +1,309 @@
+import { render, screen, within } from '@testing-library/react'
+import { PropsTable, type ComponentProp } from '../PropsTable'
+
+const componentName = 'TestComponent'
+const componentDescription =
+  'This is the testable component for the PropsTable.'
+const propWithAllTableColumns: ComponentProp = {
+  name: 'propWithAllTableColumns',
+  type: 'string | () => void',
+  defaultValue: 'foobarbaz',
+  description: 'This prop has data for all table fields.',
+}
+const propWithNameOnly: ComponentProp = {
+  name: 'propWithNameOnly',
+}
+
+it('Renders component name in a heading level 3 by default', () => {
+  render(<PropsTable componentName={componentName} />)
+
+  expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent(
+    componentName,
+  )
+})
+
+it('Renders component name in heading level when headingLevel is passed in', () => {
+  render(<PropsTable headingLevel="h2" componentName={componentName} />)
+
+  expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
+    componentName,
+  )
+})
+
+it('Does not render component description by default', () => {
+  render(<PropsTable componentName={componentName} />)
+
+  expect(screen.queryByTestId('component-description')).not.toBeInTheDocument()
+})
+
+it('Renders component description when componentDescription is passed in', () => {
+  render(
+    <PropsTable
+      componentName={componentName}
+      componentDescription={componentDescription}
+    />,
+  )
+
+  expect(screen.getByText(componentDescription)).toBeVisible()
```

Comment:
Nit: if you're using the testId to check that the description isn't rendering in the last test I would expect an assertion that it is rendering in this test.

Up to you whether you have both assertions or assert against the text content of the queried test id.